apps/opik-backend/src/main/java/com/comet/opik/domain/PromptDAO.java

```diff
@@ -22,4 +26,20 @@ interface PromptDAO {
     @SqlQuery("SELECT * FROM prompts WHERE id = :id AND workspace_id = :workspaceId")
     Prompt findById(@Bind("id") UUID id, @Bind("workspaceId") String workspaceId);
 
+    @SqlQuery("SELECT * FROM prompts " +
+            " WHERE workspace_id = :workspace_Id " +
+            " <if(name)> AND name like concat('%', :name, '%') <endif> " +
+            " ORDER BY id DESC " +
+            " LIMIT :limit OFFSET :offset ")
+    @UseStringTemplateEngine
+    @AllowUnusedBindings
+    List<Prompt> find(@Define("name") @Bind("name") String name, @Bind("workspace_Id") String workspaceId,
+            @Bind("offset") int offset, @Bind("limit") int limit);
+
+    @SqlQuery("SELECT COUNT(id) FROM prompts " +
+            " WHERE workspace_id = :workspace_Id " +
+            " <if(name)> AND name like concat('%', :name, '%') <endif> ")
+    @UseStringTemplateEngine
+    @AllowUnusedBindings
```

Comment:
Do we need this?

Reply:
Yeah, because of the string template. If the name is null, it will not render the where clause, and there will be no `:name` argument.